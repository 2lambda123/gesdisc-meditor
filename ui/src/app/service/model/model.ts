/**
 * Model Editor API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ModelIcon } from './modelIcon';


export interface Model {
    /**
     * Name of the model
     */
    name: string;
    /**
     * Model's description
     */
    description: string;
    icon: ModelIcon;
    /**
     * Model's schema
     */
    schema: string;
    /**
     * Model layout based on Angular JSON Schema form
     */
    layout: string;
    /**
     * Property name in the Model's schema for using as title of the document instance of the model
     */
    titleField?: string;
    /**
     * Any documentation (text) for the Model
     */
    documentation?: string;
    tag?: Array<string>;
}
