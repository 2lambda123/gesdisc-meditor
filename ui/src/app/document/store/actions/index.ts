export * from './history.actions';
export * from './document.actions';