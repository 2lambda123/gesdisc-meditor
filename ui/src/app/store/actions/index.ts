export * from './model.actions';
export * from './notification.actions';