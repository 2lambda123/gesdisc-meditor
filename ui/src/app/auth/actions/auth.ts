import { Action } from '@ngrx/store';
import { User, Authenticate } from '../models/user';

export enum AuthActionTypes {
  GetUser = '[Auth] GetUser',
  Logout = '[Auth] Logout',
  LoginSuccess = '[Auth] Login Success',
  LoginCallback = '[Auth] Login Callback',
  LoginFailure = '[Auth] Login Failure',
  LoginRedirect = '[Auth] Login Redirect',
}

export class GetUser implements Action {
  readonly type = AuthActionTypes.GetUser;
}

export class LoginSuccess implements Action {
  readonly type = AuthActionTypes.LoginSuccess;

<<<<<<< HEAD
  constructor(public payload: Object) {}
=======
  constructor(public payload: User) {}
}

export class LoginCallback implements Action {
  readonly type = AuthActionTypes.LoginCallback;

  constructor(public payload: string) {}
>>>>>>> 58c3fd25434c01efed8927bc8045b7a611cd1612
}

export class LoginFailure implements Action {
  readonly type = AuthActionTypes.LoginFailure;

  constructor(public payload: any) {}
}

export class LoginRedirect implements Action {
  readonly type = AuthActionTypes.LoginRedirect;
}

export class Logout implements Action {
  readonly type = AuthActionTypes.Logout;
}

export type AuthActions =
  | GetUser
  | LoginSuccess
  | LoginCallback
  | LoginFailure
  | LoginRedirect
  | Logout;
