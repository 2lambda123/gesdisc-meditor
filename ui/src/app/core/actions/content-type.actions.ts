import { Action } from '@ngrx/store';
import { ContentType } from '../../models/content-type';

export enum ContentTypesActionTypes {
	SelectContentType = '[ContentType] Select Content Type',
	LoadContentTypes = '[ContentType] Load content types',
	LoadComplete = '[ContentType] Load Complete',
	LoadError = '[ContentType] Load Error',
}

export class SelectContentType implements Action {
	readonly type = ContentTypesActionTypes.SelectContentType;

	constructor(public payload: string) { }
}

export class LoadContentTypes implements Action {
	readonly type = ContentTypesActionTypes.LoadContentTypes;

	constructor() { }
}

export class LoadComplete implements Action {
	readonly type = ContentTypesActionTypes.LoadComplete;

	constructor(public payload: ContentType[]) {}
}

export class LoadError implements Action {
	readonly type = ContentTypesActionTypes.LoadError;

	constructor(public payload: any) {}
}


export type ContentTypesActions =
	| SelectContentType
	| LoadContentTypes
	| LoadComplete
	| LoadError;
