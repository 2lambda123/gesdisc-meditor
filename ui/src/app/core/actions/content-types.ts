import { Action } from '@ngrx/store';
import { ContentType } from '@models/content-type';

export enum ContentTypesActionTypes {
	LoadContentTypes = '[ContentType] Add content types',
	LoadComplete = '[ContentType] Load Complete',
  LoadError = '[ContentType] Load Error',
}

export class LoadContentTypes implements Action {
  readonly type = ContentTypesActionTypes.LoadContentTypes;

  // constructor(public payload: ContentType[]) { }
}

export class LoadComplete implements Action {
  readonly type = ContentTypesActionTypes.LoadComplete;

  constructor(public payload: ContentType[]) {}
}

export class LoadError implements Action {
  readonly type = ContentTypesActionTypes.LoadError;

  constructor(public payload: string) {}
}


export type ContentTypesActions =
	LoadContentTypes
	| LoadComplete
	| LoadError;
