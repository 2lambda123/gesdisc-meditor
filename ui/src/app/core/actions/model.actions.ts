import { Action } from '@ngrx/store';
import { Model } from '../../service/model/model';

export enum ModelActionTypes {
	SelectModel = '[Model] Select Content Type',
	Load = '[Model] Load content types',
	LoadComplete = '[Model] Load Complete',
	LoadError = '[Model] Load Error',
}

export class SelectModel implements Action {
	readonly type = ModelActionTypes.SelectModel;

	constructor(public payload: string) { }
}

export class Load implements Action {
	readonly type = ModelActionTypes.Load;

	constructor() { }
}

export class LoadComplete implements Action {
	readonly type = ModelActionTypes.LoadComplete;

	constructor(public payload: Model[]) {}
}

export class LoadError implements Action {
	readonly type = ModelActionTypes.LoadError;

	constructor(public payload: any) {}
}


export type ModelActionsUnion =
	| SelectModel
	| Load
	| LoadComplete
	| LoadError;
