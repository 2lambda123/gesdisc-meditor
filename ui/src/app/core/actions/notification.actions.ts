import { MatSnackBarConfig } from "@angular/material";
import { Action } from '@ngrx/store';

export enum NotificationActionTypes {
	NotificationOpen = '[Notification] Open notificaton',
	NotificationClose = '[Notification] Close notificaton'
}

export class NotificationOpen implements Action {
  readonly type = NotificationActionTypes.NotificationOpen;

  constructor(public payload: {
    message: string,
    action?: string,
    config?: string
  }) { }

}

export class NotificationClose implements Action {
  readonly type = NotificationActionTypes.NotificationClose;
}


export type NotificationActionsUnion =
	| NotificationOpen
	| NotificationClose;
