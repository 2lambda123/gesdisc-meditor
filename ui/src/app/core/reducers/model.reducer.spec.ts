import { reducer } from './model.reducer';
import * as fromModels from './model.reducer';
import { LoadComplete, Load, SelectModel } from '../actions/model.actions';
import { Model } from '../../service/model/model';

describe('ModelsReducer', () => {
	//const book1 = generateMockBook();
	const model1 = { name: 'testModel', description: 'testDescription'};
	const model2 = { ...model1, name: '222' };
	const model3 = { ...model2, name: '333' };
	const initialState: fromModels.State = {
		ids: [model1.name, model2.name],
		entities: {
			[model1.name]: model1,
			[model2.name]: model2,
		},
		selectedModelId: 'Alerts',
	};

	describe('undefined action', () => {
		it('should return the default state', () => {
			const result = reducer(undefined, {} as any);

			expect(result).toMatchSnapshot();
		});
	});

	describe('Load Complete', () => {
		function noExistingModels(
			action: any,
			modelsInitialState: any,
			initialState: any,
			models: Model[]
		) {
			const createAction = new action(models);

			const result = reducer(modelsInitialState, createAction);

			expect(result).toMatchSnapshot();
		}

		function existingModels(action: any, initialState: any, models: Model[]) {
			// should not replace existing models
			const differentModel2 = { ...models[0], foo: 'bar' };
			const createAction = new action([models[1], differentModel2]);

			const expectedResult = {
				ids: [...initialState.ids, models[1].name],
				entities: {
					...initialState.entities,
					[models[1].name]: models[1],
				},
				selectedModelId: null,
			};

			const result = reducer(initialState, createAction);

			expect(result).toMatchSnapshot();
		}

		it('should add all models in the payload when none exist', () => {
			noExistingModels(LoadComplete, fromModels.initialState, initialState, [
				model1,
				model2,
			]);

		});

		it('should add only new models when models already exist', () => {
			existingModels(LoadComplete, initialState, [model2, model3]);
		});
	});

	describe('Select Model', () => {
		it('should set the selected model id on the state', () => {
			const action = new SelectModel(model1.name);

			const result = reducer(initialState, action);

			expect(result).toMatchSnapshot();
		});
	});

	describe('Selectors', () => {
		describe('getSelectedId', () => {
			it('should return the selected id', () => {
				const result = fromModels.getSelectedId({
					...initialState,
					selectedModelId: model1.name,
				});

				expect(result).toMatchSnapshot();
			});
		});
	});
});
