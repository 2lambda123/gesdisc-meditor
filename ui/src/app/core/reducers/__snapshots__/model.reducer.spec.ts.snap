// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelsReducer Load Complete should add all models in the payload when none exist 1`] = `
Object {
  "entities": Object {
    "222": Object {
      "description": "testDescription",
      "name": "222",
    },
    "testModel": Object {
      "description": "testDescription",
      "name": "testModel",
    },
  },
  "ids": Array [
    "testModel",
    "222",
  ],
  "selectedModelId": "Alerts",
}
`;

exports[`ModelsReducer Load Complete should add only new models when models already exist 1`] = `
Object {
  "entities": Object {
    "222": Object {
      "description": "testDescription",
      "name": "222",
    },
    "333": Object {
      "description": "testDescription",
      "name": "333",
    },
    "testModel": Object {
      "description": "testDescription",
      "name": "testModel",
    },
  },
  "ids": Array [
    "testModel",
    "222",
    "333",
  ],
  "selectedModelId": "Alerts",
}
`;

exports[`ModelsReducer Select Model should set the selected model id on the state 1`] = `
Object {
  "entities": Object {
    "222": Object {
      "description": "testDescription",
      "name": "222",
    },
    "testModel": Object {
      "description": "testDescription",
      "name": "testModel",
    },
  },
  "ids": Array [
    "testModel",
    "222",
  ],
  "selectedModelId": "testModel",
}
`;

exports[`ModelsReducer Selectors getSelectedId should return the selected id 1`] = `"testModel"`;

exports[`ModelsReducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "selectedModelId": "Alerts",
}
`;
